syntax = "proto3";

import public "process.proto";

package dyntrace.proto.command;

message process_request
{
    oneof process
    {
        int32 pid = 1;
        string name = 2;
    }
    dyntrace.proto.process.request req = 3;
}

message list_process
{

}

message request
{
    oneof req
    {
        process_request to_proc = 1;
        list_process list_proc = 2;
    }
}

message command_message
{
    uint64 seq = 1;
    oneof body
    {
        request req = 2;
        dyntrace.proto.response resp = 3;
    }
}